base/src/main/java/io/spine/base/Environment.java

```diff
@@ -69,101 +108,65 @@ public Environment createCopy() {
     }
 
     /**
-     * Restores the state from the instance created by {@link #createCopy()}.
+     * Determines the current environment type.
      *
-     * <p>Call this method when cleaning up tests that modify {@code Environment}.
-     */
-    @VisibleForTesting
-    public void restoreFrom(Environment copy) {
-        // Make sure this matches the set of fields copied in the copy constructor.
-        this.tests = copy.tests;
-    }
-
-    /**
-     * Verifies if the code currently runs under a unit testing framework.
+     * <p>If {@linkplain #registerCustom(Class) user has defined custom env types}, goes through
+     * them in an undefined order. Then, checks the {@linkplain BaseEnvironmentType base env
+     * types}.
      *
-     * <p>The method returns {@code true} if the following packages are discovered
-     * in the stacktrace:
-     * <ul>
-     *     <li>{@code org.junit}
-     *     <li>{@code org.testng}
-     * </ul>
+     * <p> Note that if all of the {@link EnvironmentType#currentlyOn()} checks have returned
+     * {@code false}, this method falls back on {@link BaseEnvironmentType#PRODUCTION}.
      *
-     * @return {@code true} if the code runs under a testing framework, {@code false} otherwise
+     * @return the current environment type.
      */
-    @SuppressWarnings({
-            "DynamicRegexReplaceableByCompiledPattern", // OK as we cache the result
-            "DuplicateStringLiteralInspection" // used in another context
-    })
-    public boolean isTests() {
```

Comment:
Let's bring this public API method back. Its implementation will be simpler, that's for sure.